src/algorithms.rs

```diff
@@ -80,6 +81,86 @@ impl Algorithm {
             Algorithm::EdDSA => AlgorithmFamily::Ed,
         }
     }
+
+    /// Converting Key Algorithm to Algorithm
+    pub fn frm_key_alogorithm(s: &KeyAlgorithm) -> Result<Self> {
+        Algorithm::from_str(s.to_string().as_str())
+    }
+}
+
+/// The algorithms of the keys
+#[allow(clippy::upper_case_acronyms)]
+#[derive(Debug, PartialEq, Eq, Hash, Copy, Clone, Serialize, Deserialize)]
+pub enum KeyAlgorithm {
```

Comment:
should this leave in the jwk file since it's only used there?

Reply:
But isn't it better be inside `algorithms` file? That will keep both enums visually in check.

Reply:
Not really I think. I'd rather have a kinda self-contained jwk.rs rather than having to see definitions in various places

Reply:
Sure. Let's move it.